### ISSUE TITLE
Bearer token with identify scope panics on get_current_user()

### ISSUE THREAD

**Issue by AhiVT** [serenity-rs/serenity#740]
A Bearer token type Serenity Http client will thread panic when calling `client.get_current_user()` if the scope does not include `identify email`. The error returned by Serenity is `Json(Error("missing field 'verified'))` The fields `verified` and `email` are not returned with the `CurrentUser` object if the `email` oauth2 scope is omitted.

**Comment by AhiVT** [comment]
Repro is easy with following code snippet:
Ensure token oauth2 grant is `identify` only.
```rs
use serenity::http::Http as Discord;
// Create a raw Serenity client
let client = Discord::new_with_token(format!("Bearer {}", &res.access_token).as_str());
println!("Connected to Discord");
let user = client.get_current_user().unwrap(); // <=== Thread panic here
```

**Comment by LeSeulArtichaut** [comment]
I'd like to work on it. This seems like an easy issue, will be great for a first contribution :D
Simply wrapping the field into an `Option` will fix the problem. However this will break existing code...

**Comment by AhiVT** [comment]
Fix should be merged in with 0.8 then

**Comment by Flat** [comment]
Fixed by #746
